lib/mocha.js

```diff
@@ -482,21 +509,32 @@ Mocha.prototype.run = function(fn) {
   if (options.globals) {
     runner.globals(options.globals);
   }
+  // Use only the first reporter for growl, since we don't want to
+  // send several notifications for the same test suite
   if (options.growl) {
-    this._growl(runner, reporter);
+    this._growl(runner, reporters[0]);
   }
   if (options.useColors !== undefined) {
     exports.reporters.Base.useColors = options.useColors;
   }
   exports.reporters.Base.inlineDiffs = options.useInlineDiffs;
 
-  function done(failures) {
-    if (reporter.done) {
-      reporter.done(failures, fn);
-    } else {
-      fn && fn(failures);
+  function runnerDone(failures) {
+    function reporterDone(failures) {
+      if (--remain === 0) {
+        fn && fn(failures);
+      }
     }
+
+    var remain = reporters.length;
+    reporters.forEach(function(reporter) {
+      if (reporter.done) {
+        reporter.done(failures, reporterDone);
+      } else {
+        reporterDone(failures);
```

Comment:
It's on scope, but the `failures` passed by `reporter.done` might be a different object. Contrived example:

`reporter` is a reporter with the following `done` implementation:
```javascript
done(failures, cb) {
   cb(failures.concat({}))
}
```

If `reporterDone` used the `failures` captured before calling `reporter.done`, the above code would't work. 

I'm not sure it's a real case, but I think it's safer, and the as far as I know the callback signature is `(failures)`, so I think we should reflect & honor that there and use the actual object passed by arg.